src/CSET/operators/regrid.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2024 Met Office and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Operators to regrid cubes."""
+
+import iris
+import iris.cube
+import numpy as np
+
+
+def regrid_onto_cube(
+    incube: iris.cube.Cube, target: iris.cube.Cube, method: str, **kwargs
+) -> iris.cube.Cube:
+    """Regrid a cube, projecting onto a target cube.
+
+    Cube must have at least 2 dimensions.
+
+    Arguments
+    ----------
+    incube: Cube
+        An iris cube of the data to regrid. As a minimum, it needs to be 2D with a latitude,
+        longitude coordinates.
+    target: Cube
+        An iris cube of the data to regrid onto. It needs to be 2D with a latitude,
+        longitude coordinate.
+    method: str
+        Method used to regrid onto, etc. Linear will use iris.analysis.Linear()
+
+    Returns
+    -------
+    iris.cube.Cube
+        An iris cube of the data that has been regridded.
+
+    Raises
+    ------
+    ValueError
+        If a unique x/y coordinate cannot be found
+    NotImplementedError
+        If the cubes grid, or the method for regridding, is not yet supported.
+
+    Notes
+    -----
+    The acceptable coordinate names for X and Y coordinates are currently described
+    in X_COORD_NAMES and Y_COORD_NAMES. These cover commonly used coordinate types,
+    though a user can append new ones.
+    Currently rectlinear grids (uniform) are supported.
+    """
+    # Usual names for spatial coordinates
+    # TODO can we determine grid coord names in a more intelligent way?
+    X_COORD_NAMES = ["longitude", "grid_longitude", "projection_x_coordinate", "x"]
+    Y_COORD_NAMES = ["latitude", "grid_latitude", "projection_y_coordinate", "y"]
```

Comment:
If you move these constant definitions out of the function they will be avaliable for the whole module. I.e. you won't have to define them in both onto_cube and onto_xyspacing.

Reply:
Good point, now made them global